### PR TITLE
BIN-1012-Navadian

### PR DESCRIPTION
**Pull request by fausto-focus**
Add the SPI protocol API example
# Resolves

- https://focusuy.atlassian.net/browse/BIN-1012

# Changes

- Creates a notebook section for the Supernova SDK tests to adapt to the structure already created in BIN-991
- Adds the SPI protocol API example
- This example includes tests of the different commands related to the Supernova SPI module to show the feature's capabilities. It also includes an example of the Supernova interfacing with the Adafruit SPI FRAM, performing a write and read operation.

# How to test

- To test the commands, connect the breakout board to the Supernova and the logic analyzer in the correspondent SPI signals to verify the correct operation
- Follow step by step the explanation in the SPI-Protocol-Supernova-API.ipynb juypter notebook
- In the **Example with FRAM memory MB85RS64V** section connect the signals from the breakout board to its correspondent signals in the FRAM memory
- Execute the rest of the cells of the SPI-Protocol-Supernova-API.ipynb juypter notebook

### CODE REVIEW

File: notebooks/SPI-Protocol-API/SPI-Protocol-Supernova-API.ipynb [hunk 1]
```diff
@@ -0,0 +1,1440 @@
...
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SPI Protocol API\n",
+    "\n",
+    "The SPI Protocol API methods are described below.\n",
+    "\n",
+    "The first part of the demo presents all the commands allowed to use in the SPI Protocol API, showing different examples of the capabilities of this feature. The transfers and configurations utilized in this first part may not be compatible with the memory used in the second part. Having this in consideration, a suggestion for the exploration of the commands and testing is not to use any SPI target connected and use a logic analyzer instead to notice everything is working fine."
```
Comment (NicoleCabotImperial) [inline notebooks/SPI-Protocol-API/SPI-Protocol-Supernova-API.ipynb:222]: and use a sniffer instead to verify*

**Review by NicoleCabotImperial — APPROVED** [review]
Approved 💯

**Review by Salonso1602 — CHANGES_REQUESTED** [review]
Great work! 
Ive left some small comments

**Review by Salonso1602 — CHANGES_REQUESTED** [review]
I think these are my final suggestions, great work!

**Pull request merged**